src/v3/src/hooks/useOnSubmit.ts

```diff
@@ -100,8 +101,21 @@ export const useOnSubmit = (): (options: OnSubmitHandlerOptions) => Promise<void
     try {
       const newTransaction = await fn(payload);
       setIdxTransaction(newTransaction);
-      const isClientTransaction = newTransaction.nextStep?.name === currTransaction?.nextStep?.name;
+      const isClientTransaction = !(newTransaction.requestDidSucceed ?? true);
       setIsClientTransaction(isClientTransaction);
+      if (isClientTransaction
+          && !newTransaction.messages?.length
```

Comment:
Messages from the transaction are added to the ui schema during the normal transformation flow, so here we need to avoid displaying the generic global error message and the transaction errors.